### PR TITLE
CLOUDSTACK-10247: L2 network not shared on projects

### PR DESCRIPTION
**Pull request by nvazquez**
When trying to deploy a vm providing a project id and a L2 network id, this error is logged:

````
2018-01-18 09:04:31,749 INFO  [c.c.a.ApiServer] (qtp1310540333-17:ctx-7ff91c56 ctx-8c37c46b ctx-d58064ad) (logid:74e86028) PermissionDenied: Unable to use network with id= 5bee486a-ff20-4db2-b62e-4b4f3485cfff, permission denied on objs: []
````

### CODE REVIEW

**Comment by borisstoyanov** [comment]
@blueorangutan package

**Comment by borisstoyanov** [comment]
@blueorangutan test

**Comment by nvazquez** [comment]
Thanks @borisstoyanov! @DaanHoogland @rhtyd can you review?

File: server/src/com/cloud/network/NetworkModelImpl.java [hunk 1]
```diff
@@ -1637,7 +1637,8 @@ public void checkNetworkPermissions(Account owner, Network network) {
             throw new CloudRuntimeException("cannot check permissions on (Network) <null>");
         }
         // Perform account permission check
-        if (network.getGuestType() != Network.GuestType.Shared || (network.getGuestType() == Network.GuestType.Shared && network.getAclType() == ACLType.Account)) {
+        if ((network.getGuestType() != Network.GuestType.Shared && network.getGuestType() != GuestType.L2) ||
```
Comment (DaanHoogland) [inline server/src/com/cloud/network/NetworkModelImpl.java]: Network.GuestType.Shared is used and then GuestType.L2 directly. please cleanup the imports and use only one scope for one java class.

**Comment by nvazquez** [comment]
Thanks @DaanHoogland, there were some other occurrences of `Network.GuestType` and `GuestType`, last commit unifies all occurrences to `GuestType`

**Comment by rhtyd** [comment]
LGTM, merging based on reviews and test results.

**Pull request merged**